src/frontend/src/lib/types/dapp.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod';
+
+// see https://github.com/dfinity/portal/tree/95c67a5cfe201e4e5cb79f3cf5d18fe16498cd8c?tab=readme-ov-file#object-schema
+const dAppSchema = z.object({
+	id: z.string(),
+	name: z.string(),
+	oneLiner: z.string(),
```

Comment:
There seems to be some sort of min and max logic implemented in the website. We should try to replicate those rules. Does not need to be in this PR so maybe you can add a TODO referencing the code of the portal: https://github.com/dfinity/portal/blob/34a0328ed4792f5a7f3943be73f13f5abaefb4b8/plugins/validate-showcase.js#L179